src/realm.js

```diff
@@ -271,6 +282,45 @@ export class Realm {
     }
   }
 
+  add_prior_effects(priorEffects: Effects, subsequentEffects: Effects) {
+    let [pc, pg, pb, pp, po] = priorEffects;
+    let [sc, sg, sb, sp, so] = subsequentEffects;
+
+    pc; sc;
+
+    let saved_generator = this.generator;
+    this.generator = pg.clone();
```

Comment:
Why do you clone `pg` here? It seems like you build up a new generator that then gets thrown away